### ISSUE TITLE
EKS Calico to prevent ip exhaustion

### ISSUE THREAD

**Issue by guzzijones** [StackStorm/stackstorm-k8s#361]
# TLDR
nodes end up with 1 ip address attached rather than an ip for each pod

# Instructions
Calico is a better alternative to VPC-CNI if you want to avoid using up all the ip addresses in a subnet.  VPC-CNI assigns a subnet IP address to each pod making them first class citizens.  Calico creates a network layer and handles the ips itself.  

To accomplish this it wasn't too bad:

1.  create an overriding chart for the tigera-operator
```
mkdir calico
cd calico
vim Chart.yaml
```
2. Chart.yaml contents
```
apiVersion: v2
name: calico
description: A Helm chart for Kubernetes
type: application
version: 0.1.0
appVersion: "0.1.0"
dependencies:
  - name: tigera-operator
    version: 3.25.0
    repository: https://docs.tigera.io/calico/charts/
```
3. create values.yaml (override registry setting with your registry as below  if on internal network)
```
tigera-operator:
  installation:
    kubernetesProvider: EKS
  tigeraOperator:
    image: tigera/operator
    version: v1.29.0
    registry: myregistry.io/ext.quay.io/
  calicoctl:
    image:  myregistry.io/docker.io/calico/ctl
    tag: v3.25.0
  registry: myregistry.io/ext.docker.io/
```
4. make a manifest file for your calico networking settings operator.yaml  (override registry as below with your registry if on internal network)
```
kind: Installation
apiVersion: operator.tigera.io/v1
metadata:
  name: default
spec:
  kubernetesProvider: EKS
  cni:
    type: Calico
  calicoNetwork:
    bgp: Disabled
  registry: myregistry.com/ext.docker.io/
```
4. install helm operator and settings
```
~/kubectl create namespace tigera-operator
helm dependency build
helm install tigera-operator . -n tigera-operator
kubectl apply -f operator.yaml
```

# Tigera Docs

[docs](https://docs.tigera.io/calico/3.25/getting-started/kubernetes/managed-public-cloud/eks)

# installation settings spec

[tigera installation settings ](https://docs.tigera.io/calico/3.25/reference/installation/api#operator.tigera.io/v1.Installation)
